src/cluster_migrateslots.c

```diff
@@ -0,0 +1,2119 @@
+ *         └──────────────────┬────────────────┘   │
+ *    Full response read (+OK)│                    │
+ *           ┌────────────────▼──────────────┐     │ Error Conditions:
+ *           │SLOT_EXPORT_WAITING_TO_SNAPSHOT┼─────┤  1. User sends CANCELMIGRATIONS
+ *           └────────────────┬──────────────┘     │  2. Slot ownership change
+ *      No other child process│                    │  3. Demotion to replica
+ *               ┌────────────▼───────────┐        │  4. FLUSHDB
+ *               │SLOT_EXPORT_SNAPSHOTTING┼────────┤  5. Connection Lost
+ *               └────────────┬───────────┘        │  6. AUTH failed
+ *               Snapshot done│                    │  7. ERR from ESTABLISH command
+ *                ┌───────────▼─────────┐          │  8. Unpaused before failover completed
+ *                │SLOT_EXPORT_STREAMING┼──────────┤  9. Snapshot failed (e.g. Child OOM)
+ *                └───────────┬─────────┘          │  10. No ack from target (timeout)
+ *               REQUEST-PAUSE│                    │  11. Client output buffer overrun
+ *             ┌──────────────▼─────────────┐      │
+ *             │SLOT_EXPORT_WAITING_TO_PAUSE┼──────┤
+ *             └──────────────┬─────────────┘      │
+ *              Buffer drained│                    │
+ *             ┌──────────────▼────────────┐       │
+ *             │SLOT_EXPORT_FAILOVER_PAUSED┼───────┤
+ *             └──────────────┬────────────┘       │
+ *    Failover request granted│                    │
+ *            ┌───────────────▼────────────┐       │
+ *            │SLOT_EXPORT_FAILOVER_GRANTED┼───────┤
+ *            └───────────────┬────────────┘       │
+ *       New topology received│                    │
+ *             ┌──────────────▼────────────┐       │
+ *             │SLOT_MIGRATION_JOB_SUCCESS│       │
+ *             └───────────────────────────┘       │
+ *                                                 │
+ *             ┌──────────────────────────┐        │
+ *             │SLOT_MIGRATION_JOB_FAILED│◄───────┤
+ *             └──────────────────────────┘        │
+ *                                                 │
+ *            ┌─────────────────────────────┐      │
+ *            │SLOT_MIGRATION_JOB_CANCELLED│◄─────┘
+ *            └─────────────────────────────┘
+ *
+ */
+
+/* Returns 1 if the given slot is being exported, 0 otherwise. */
+int clusterIsSlotExporting(int slot) {
+    listNode *ln;
+    listIter li;
+    listRewind(server.cluster->slot_migration_jobs, &li);
+    while ((ln = listNext(&li)) != NULL) {
+        slotMigrationJob *job = ln->value;
+        if (job->type != SLOT_MIGRATION_EXPORT) continue;
+        if (isSlotMigrationJobFinished(job)) continue;
+        if (isSlotInSlotRanges(slot, job->slot_ranges)) return 1;
+    }
+    return 0;
+}
+
+/* Returns the name of the slot export target for the given slot, or NULL if the
+ * slot is not being exported. */
+char *getNameOfSlotExportTarget(int slot) {
```

Comment:
nit - constness 
```suggestion
const char *getSlotExportTargetName(int slot) {
```

btw, I still feel that `getSlotExportTargetName` reads better

Reply:
Yeah - I updated the function so it does the logic it so it no longer returns a char* but a bool